obspy/core/trace.py

```diff
@@ -158,7 +158,10 @@ def __setitem__(self, key, value):
             # ensure correct data type
             if key == 'delta':
                 key = 'sampling_rate'
-                value = 1.0 / float(value)
+                if value > 0.0:
+                    value = 1.0 / float(value)
+                else:
+                    value = 0.0
```

Comment:
Can you use the same `try/except` construct that is used for this check for the sampling rate a couple of lines down? Or change the other one as well? Better to have things consistent.

Reply:
I'll go with `try/except` on zerodivision. This will let whatever the behavior of negative SR and deltas to remain.